lib/core/src/data/weakset.rs

```diff
@@ -0,0 +1,113 @@
+use std::collections::HashSet;
+use std::rc::{Rc, Weak};
+use std::hash::{Hash, Hasher};
+pub trait Hashable = Hash + PartialEq;
```

Comment:
There should be 2 empty lines between imports and code + please, use the code separators:

```
// ================
// === Hashable ===
// ================
```
to divide parts of the file related to different entities. If there is a single entitiy, we use just a single separator on top of the file.